`numpy>=2.4.0` introduces type errors in the tests.

**Describe the bug**


See [numpy release notes](https://numpy.org/doc/2.4/release/2.4.0-notes.html#raise-typeerror-on-attempt-to-convert-array-with-ndim-0-to-scalar)

**To Reproduce**
Steps to reproduce the behavior/error:
1. have numpy>=2.4.0
2. run the tests (e.g. `climada/entity/exposures/test/test_mat.py`)
3. Observe the failure:

```
climada/entity/exposures/test/test_mat.py:37: in test_read_demo_pass
    expo = Exposures.from_mat(ENT_TEST_MAT)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
climada/entity/exposures/base.py:1277: in from_mat
    _read_mat_metadata(exp, data, file_name, var_names)
climada/entity/exposures/base.py:1597: in _read_mat_metadata
    exposures.value_unit = u_hdf5.get_str_from_ref(
climada/util/hdf5_handler.py:110: in get_str_from_ref
    return get_string(file[var])
           ^^^^^^^^^^^^^^^^^^^^^
climada/util/hdf5_handler.py:92: in get_string
    return "".join(chr(int(c)) for c in array)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
climada/util/hdf5_handler.py:92: in <genexpr>
    return "".join(chr(int(c)) for c in array)
                       ^^^^^^
E   TypeError: only 0-dimensional arrays can be converted to Python scalars
```

**Expected behavior**
No failure

**Suggestion**

Numpy 2.4 now requires to specifically access an item to enable conversion to a number. Implement this change.

--- a/climada/hazard/tc_tracks_synth.py
+++ b/climada/hazard/tc_tracks_synth.py
@@ -1049,7 +1049,7 @@ def _apply_decay_coeffs(track, v_rel, p_rel, land_geom, s_rel):
                 # if there is no further landfall, correct until the end of
                 # the track
                 end_cor = track["time"].size
-            rndn = 0.1 * float(np.abs(np.random.normal(size=1) * 5) + 6)
+            rndn = 0.1 * float(np.abs(np.random.default_rng().normal() * 5) + 6)
             r_diff = (
                 track["central_pressure"][land_sea].values
                 - track["central_pressure"][land_sea - 1].values

--- a/climada/util/hdf5_handler.py
+++ b/climada/util/hdf5_handler.py
@@ -89,7 +89,7 @@ def get_string(array):
     -------
     string
     """
-    return "".join(chr(int(c[0])) for c in array)
+    return "".join(chr(int(c.item())) for c in array)
 
 
 def get_str_from_ref(file_name, var):

--- a/climada/util/hdf5_handler.py
+++ b/climada/util/hdf5_handler.py
@@ -89,7 +89,7 @@ def get_string(array):
     -------
     string
     """
-    return "".join(chr(int(c)) for c in array)
+    return "".join(chr(int(c[0])) for c in array)
 
 
 def get_str_from_ref(file_name, var):

--- a/climada/util/hdf5_handler.py
+++ b/climada/util/hdf5_handler.py
@@ -89,7 +89,7 @@ def get_string(array):
     -------
     string
     """
-    return "".join(chr(int(c.item())) for c in array)
+    return "".join(chr(c.item()) for c in array)
 
 
 def get_str_from_ref(file_name, var):

